source/main/physics/RigSpawner.cpp

```diff
@@ -6983,6 +6983,12 @@ Ogre::MaterialPtr ActorSpawner::CreateSimpleMaterial(Ogre::ColourValue color)
 
 void ActorSpawner::SetupNewEntity(Ogre::Entity* ent, Ogre::ColourValue simple_color)
 {
+    if (ent == nullptr)
+    {
+        AddMessage(Message::TYPE_WARNING, "Failed to create entity: continuing without it ...");

```

Comment:
🤢 Uhh, seriously? Don't be like TDev. If an entity fails to create, then exclude the whole Gfx object that owns it (prop_t, GfxActor::Rod or whatever). If a code needs to check `entity != nullptr` on runtime, it's legacy code.

Reply:
:blush: I should have written a comment. It is temporary until we know what is going on. I've already added it to the TODO list: https://github.com/RigsOfRods/rigs-of-rods/pull/1537#issue-209431214

Reply:
I'll remove this commit and update the Github issue ticket.

Reply:
I know, and it's completely OK to commit such code as long as you comment it `//TODO: Hack until blah blah ... Github-links ~ Ulteq, 09/2018` - that way, any casual reader will know we're not 🐷, just doing right things step by step 😃